security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -521,6 +523,21 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 		certChain = append(certChain, []byte(c)...)
 	}
 
+	// Cert exipre time by default is createTime + sc.configOptions.SecretTTL.
+	expireTime := t.Add(sc.configOptions.SecretTTL)
+	block, _ := pem.Decode(certChain)
+	if block != nil {
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err == nil {
+			expireTime = cert.NotAfter
+			log.Debugf("Cert expire time is %v", expireTime)
+		} else {
+			log.Debugf("Failed to parse certificate %v, use default cert TTL", err.Error())
+		}
+	} else {
+		log.Debug("Failed to decode certificate, use default cert TTL")
```

Comment:
Do we need any extra actions if we can't get a valid expTime from a cert? Apparently we can't use this cert for service-to-service communication any more.

Reply:
current logic is CA respect secret TTL passed from nodeagent and uses it as cert TTL; this PR is handle CA that don't respect the TTL passed to it, and we fallback to default configured TTL if cert parse fail(for some reason)

Reply:
I think the question is more on cert itself (not TTL). I assume we are receiving a cert without a valid TTL in this case. Then should we still let envoy to use this cert for mTLS communication?